tools/versioner/cmd/root.go

```diff
@@ -233,8 +231,23 @@ func updateGoModVer(goMod io.ReadWriteSeeker, newVer string) error {
 }
 
 func writeChangelog(stage string, mod modinfo.Provider) error {
-	// TODO
-	return nil
+	// don't write a changelog for a new module
+	if mod.NewModule() {
+		return nil
+	}
+	const changeLogName = "CHANGELOG.md"
+	rpt := mod.GenerateReport()
+	log, err := os.Create(filepath.Join(stage, changeLogName))
+	if err != nil {
+		return fmt.Errorf("failed to create %s: %v", changeLogName, err)
+	}
+	defer log.Close()
+	if rpt.IsEmpty() {
+		_, err = log.WriteString("No changes to exported content compared to the previous release.\n")
+		return err
```

Comment:
in what case would this happen ? and do we want it to be an error ?

Reply:
Can't be an error as this is legit.  Most common case is documentation changes (i.e. comments).  Next case would be a generator change the modifies code bodies.